### PR TITLE
use sapphire fetch wrapper on node-fetch

### PR DESCRIPTION
**Pull request by patuwwy**
- @sapphite/fetch should allow to use api-clients in browser
- ClientUtils class extracted to separate module
- Typed responses:
```
(method) HostClient.listSequences(): Promise<STHRestAPI.GetSequencesResponse>
(method) HostClient.listInstances(): Promise<STHRestAPI.GetInstancesResponse>
(method) HostClient.getLogStream(): Promise<Stream | ReadableStream>
(method) HostClient.getSequence(sequenceId: string): Promise<STHRestAPI.GetSequenceResponse>
(method) HostClient.deleteSequence(sequenceId: string): Promise<STHRestAPI.DeleteSequenceResponse>

and more
```

Actual usage: 
```
const sequenceName = (await hostClient.getSequence()).data!!.config
```
```data``` is ```any```, ```config``` is ```any```

New usage:
```
const sequenceName = await hostClient.getSequence(seq_id).config
```
```config``` is ```SequenceConfig```

or 

Actual usage: 
```
(await instance.getStream("log")).data?.pipe(process.stdout);
```

New usage:
```
(await instance.getStream("log")).pipe(process.stdout);
```

### CODE REVIEW

**Review by MichalCz — APPROVED** [review]

File: bdd/step-definitions/e2e/cli.ts [hunk 1]
```diff
@@ -181,11 +181,14 @@ Then("I get instance id", function() {
 
 Then("I kill instance", async function() {
     stdio = await getStreamsFromSpawn("/usr/bin/env", [...si, "inst", "kill", instanceId, ...formatFlags(), ...connectionFlags()]);
+
     assert.equal(stdio[2], 0);
 });
 
 Then("I delete sequence", { timeout: 10000 }, async function() {
     stdio = await getStreamsFromSpawn("/usr/bin/env", [...si, "seq", "delete", sequenceId, ...formatFlags(), ...connectionFlags()]);
+    console.log(stdio);
```
[the remark this answers is not in the document]
Comment (patuwwy) [reply bdd/step-definitions/e2e/cli.ts]: removed

File: bdd/step-definitions/e2e/cli.ts [hunk 2]
```diff
@@ -238,25 +241,20 @@ Then("I get list of sequences", async function() {
...
-    assert.equal(stdio[2], 0);
-    const instances = JSON.parse(stdio[0].replace("\n", ""));
 
-    let instanceFound = false;
+    assert.equal(stdio[2], 0);
+    const instances = JSON.parse(stdio[0].replace("\n", "")) as STHRestAPI.GetInstancesResponse;
 
-    for (let i = 0; i < instances.length; i++) {
-        const inst = instances[i];
+    const instanceFound = instances.some(({ id }) => id === instanceId);
```
Comment (alicja-gruzdz) [inline bdd/step-definitions/e2e/cli.ts:248]: I always learn something new about js :)

File: bdd/step-definitions/hub/config.ts [hunk 3]
```diff
@@ -111,13 +112,10 @@ Then("send fake stream as sequence", async function(this: CustomWorld) {
...
-    try {
-        this.resources.sequenceSendPromise = hostClient.sendSequence(
-            this.resources.pkgFake as unknown as ReadStream
-        ).catch(() => undefined);
-    } catch {
-        /* don't care */
-    }
+    this.resources.sequenceSendPromise = hostClient.sendSequence(
+        this.resources.pkgFake as unknown as ReadStream
+    // eslint-disable-next-line no-console
```
Comment (patuwwy) [inline bdd/step-definitions/hub/config.ts:117]:
```suggestion
```

File: packages/cli/src/lib/output.ts [hunk 4]
```diff
@@ -43,16 +42,16 @@ export async function displayObject(_program: Command, object: any) {
...
-        const resp = await response;
+        const resp = (await response) as unknown as Readable;
 
-        resp.data?.pipe(output);
-        return new Promise((res, rej) => resp.data?.on("finish", res).on("error", rej));
+        resp.pipe(output);
+        return new Promise((res, rej) => resp.on("finish", res).on("error", rej));
     } catch (e: any) {
-        console.error(e && e.stack || e);
+        console.error((e && e.stack) || e);
```
Comment (patuwwy) [inline packages/cli/src/lib/output.ts:54]:
```suggestion
        console.error(e && e.stack || e);
```

File: packages/client-utils/src/client-utils.ts [hunk 5]
```diff
@@ -20,13 +21,13 @@ export class ClientUtils implements HttpClient {
...
      * @param {Partial<RequestLogger>} logger Logger to set.
-    */
+     */
     public addLogger(logger: Partial<RequestLogger>) {
         this.log = {
             request: () => 0,
             ok: () => 0,
             error: () => 0,
-            ...logger
+            ...logger,
```
Comment (patuwwy) [inline packages/client-utils/src/client-utils.ts:30]:
```suggestion
            ...logger
```

File: packages/client-utils/src/client-utils.ts [hunk 6]
```diff
@@ -148,29 +141,20 @@ export class ClientUtils implements HttpClient {
...
-            end,
-            parseResponse
-        }: SendStreamOptions = {}
-    ): Promise<Response> {
+        { type = "application/octet-stream", end, parseResponse }: SendStreamOptions = {}
+    ): Promise<T> {
         const headers: Headers = {
             "content-type": type,
-            expect: "100-continue"
+            expect: "100-continue",
```
Comment (patuwwy) [inline packages/client-utils/src/client-utils.ts:151]:
```suggestion
            expect: "100-continue"
```

**Pull request merged**